packages/ui/src/FilesTable/index.tsx

```diff
@@ -0,0 +1,139 @@
+import React from "react";
+import { ActionsMenu, DataTable } from "../index";
+import { FilterMatchMode } from "primereact/api";
+import { ColumnProps } from "primereact/column";
+import { MenuItem } from "primereact/menuitem";
+
+type TranslationMessage = {
+  actionDownloadMessage: string;
+  actionEditMessage: string;
+  actionRenameMessage: string;
+  columnFileHeaderMessage: string;
+  columnDescriptionHeaderMessage: string;
+  searchPlaceholderMessage: string;
+  columnUploadedByHeaderMessage: string;
+  columnUploadedAtHeaderMessage: string;
+  columnActionsHeaderMessage: string;
+  tableEmptyMessage: string;
+};
+
+export type FilesTableProperties = {
+  className?: string;
+  columns?: Array<ColumnProps>;
+  id?: string;
+  loading?: boolean;
+  files: Array<object>;
+  totalRecords?: number;
+  showDescriptionColumn?: boolean;
+  extraColumns?: Array<ColumnProps>;
+  fetchFiles?: (arguments_?: any) => void;
```

Comment:
We may have type defined for arguments. Please check in Users or Invitaitons table.